waspc/src/Wasp/Analyzer/Parser.hs

```diff
@@ -36,11 +37,19 @@ import Control.Monad.Except (runExcept)
 import Control.Monad.State (evalStateT)
 import Wasp.Analyzer.Parser.AST
 import Wasp.Analyzer.Parser.Ctx (Ctx (..), WithCtx (..), ctxFromPos, ctxFromRgn, fromWithCtx, getCtxRgn, withCtx)
-import Wasp.Analyzer.Parser.Monad (initialState)
+import Wasp.Analyzer.Parser.Monad (makeInitialState)
 import Wasp.Analyzer.Parser.ParseError
 import qualified Wasp.Analyzer.Parser.Parser as P
 import Wasp.Analyzer.Parser.SourcePosition (SourcePosition (..))
 import Wasp.Analyzer.Parser.Token
 
 parse :: String -> Either ParseError AST
-parse = runExcept . evalStateT P.parse . initialState
+parse = runExcept . evalStateT P.parse . makeInitialState
+
+parseExpression :: String -> Either ParseError Expr
+parseExpression = runExcept . evalStateT  P.parseExpression . makeInitialState 
+
+isValidWaspIdentifier :: String -> Bool
+isValidWaspIdentifier str = case parseExpression str of
+  Right (Var name) -> let noCharsSkipped = length name == length str in noCharsSkipped
+  _ -> False
```

Comment:
This feels like it could maybe go somewhere else, but I am not sure where! Maybe in the future it will become more clear so it is ok to leave it here for now.

Reply:
Btw I never like this `Var` name, I would rather it is called smth like `Identifier`, but ok, I believe there were some reasons for naming it `Var`.

Reply:
> This feels like it could maybe go somewhere else, but I am not sure where! Maybe in the future it will become more clear so it is ok to leave it here for now.

Yes, I thought of moving it; this is still a rough version.

> Btw I never like this Var name, I would rather it is called smth like Identifier, but ok, I believe there were some reasons for naming it Var.

Not a big fan myself. I think I'll change it. The name `Identifier` is a better fit because we probably won't have different lexical rules for naming functions or constants (if we ever end up supporting them), and those are all identifiers.

Reply:
Makes sense! But as I said, I remmeber there was a reason why @craigmc08 named it Var, as I believe I already tried to argument we name it Identifier, and if we agreed to have it as Var, it was probably ok reason. Maybe it was what other compilers used? Hmmm. But ok, if we think now it is better to name it Identifier I am ok with that.